osu.Game/Overlays/Settings/SettingsItem.cs

```diff
@@ -83,8 +81,20 @@ public virtual LocalisableString LabelText
 
         public virtual Bindable<T> Current
         {
-            get => controlWithCurrent.Current;
-            set => controlWithCurrent.Current = value;
+            get
+            {
+                if (!(Control is IHasCurrentValue<T> controlWithCurrent))
+                    throw new ControlDoesNotImplementCurrentException();
+
+                return controlWithCurrent.Current;
+            }
+            set
+            {
+                if (!(Control is IHasCurrentValue<T> controlWithCurrent))
+                    throw new ControlDoesNotImplementCurrentException();
+
+                controlWithCurrent.Current = value;
+            }
```

Comment:
I do not think this change is desirable. So now, apparently, `Current` - which is *integral* to this setting stuff working properly - can throw in runtime? Because of one edge-case mod? That seems backwards.

Reply:
Not sure what your point is? I don't see how it's backwards to actually implement a `DifficultyBindable` current inside `SliderControl` itself, if it's gonna completely ignore it there and use a different bindable, that's the opposite to me.

Reply:
The point is that while this change does remove the bindable passing, it does so at the expense of willingly bypassing the core structure of how settings controls work. Which is not unequivocally better for me - more like slightly below par when compared with the current version.

The actual clean way to get rid of the bindable passing would be by having a bindable number type supporting nullable types.

Reply:
Having nullable type support that displays a certain value in case of null would make things even more clean, the way done in this PR is to have an intermediate mapping between a `DifficultyBindable` and a `BindableNumber<float>`, as in below:
https://github.com/ppy/osu/blob/cd2556bd110576e639c5ca4fc3c0c23a8bd46e6e/osu.Game/Rulesets/Mods/DifficultyAdjustSettingsControl.cs#L24-L39

And as I've mentioned, reverting back to just the simple fix makes the control in general read weirdly, but I'll wait to hear more opinions. (for example, here's how it would look if I were to apply the simple fix alone, for cross comparsion https://github.com/ppy/osu/compare/master...frenzibyte:alternative-difficulty-bindable?expand=1)

Reply:
I'm not completely against this change, but wouldn't it make more sense to move the exception / whatever to trigger after `CreateControl` is run? In fact, that's how it's done on `master`?

Reply:
But `CreateControl` is allowed to return a type that's not `IHasCurrentValue`, it's the implementation of `Current` that would decide whether to throw or not, if `Current` wasn't overriden and reached the `SettingsItem`'s implementation, then throw the exception if not implemented `IHasCurrentValue`.

Reply:
overridden current sounds like a really bad pattern in the first place.

Reply:
If I were to remove the override pattern and let `SliderControl` have the current implementation instead (`float?` one), there'll be no way to access the `DifficultyBindable` from the main control itself without fetching it from the slider control, and that feels really awkward in code readability to me. (unless we still override `Current` to save an instance of the `DifficultyBindable` before setting `base.Current`, making us jump into bindable reference issues because `Drawable` would unbind it on disposal)

Reply:
The simple variant linked above looks vastly preferable to me. I have zero issues with it and I don't follow how it would make any of the existing code read any worse _than it already does_.

If the concern is about code quality then you'd have to take a detour of a few pulls to start getting the difficulty bindable stuff in better shape. I consider the "code quality" changes in this pull mostly _orthogonal_ (i.e. some problems are "solved", but new ones are introduced in exchange).

Reply:
Are there really new problems introduced here other than the `Current` override and the allowing of having settings controls without `IHasCurrentValue`? (in exchange for overriding `Current` and connecting it to the underlying control in some form, i.e. "custom implementation")

Reply:
Allowing settings controls to not implement `IHasCurrentValue` is literally the biggest problem I have with this diff right now.

Reply:
I don't think this is gonna be easily solvable then, I'll proceed with closing this PR and going with the simple fix alone if no one has any other opinions on the matter.

Reply:
The proposed solution looks far better.